File: exemples/intro/factorielle/Factorielle2.py
# -*- encoding: utf-8 -*-
'''
    Factorielle, utilisation de la récursivité
'''
import sys

def factorielle2(n):
    if n == 0:
        return 1
    return n * factorielle2(n - 1)

def main():
    n = int(sys.argv[1])
    f = factorielle2(n)
    print f

main()

File: exemples/intro/factorielle/Factorielle2.java
/** 
    Factorielle, utilisation de la récursivité
**/
class Factorielle2 {
    public static void main(String[] args) {
        int n = (int)Integer.valueOf(args[0]);
        int f = factorielle(n);
        System.out.printf("Factorielle : %d%n", f);
    }
    
    public static int factorielle(int n){
        if(n < 2){
            return 1;
        }
        return n * factorielle(n - 1);
    }
    
}
